traffic-advice.bs

```diff
@@ -0,0 +1,265 @@
+
+    1.  Let |agentSelector| be |entry|[`"user_agent"`].
+
+    1.  If |identity| does not contain |agentSelector|, then [=iteration/continue=].
+
+    1.  If |bestMatch| is null or |agentSelector| appears at an earlier index in |identity| than |bestMatch|[`"user_agent"`], then set |bestMatch| to |entry|.
+
+1.  If |bestMatch| is null, then return null.
+
+1.  Let |entry| be a [=traffic advice entry=].
+
+1.  If |bestMatch|[`"disallowed"`] [=map/exists=] and is true, then set |entry|'s [=traffic advice entry/disallowed flag=] to true.
+
+1.  Return |entry|.
+
+</section>
+
+Interpretation {#interpretation}
+================================
+
+*This section is non-normative.*
+
+[=Traffic advice respecting agents=] should, when they would be able to respect advice to disallow traffic to an origin (for example, when requested to proxy prefetch traffic to the origin), [=fetch traffic advice=] (respecting HTTP caching semantics).
+
+If the result is null, then no advice was received and the agent should adopt its default behavior.
+
+If the result is `"unreachable"`, then the HTTP server was not able to service the request for traffic advice. The agent should disallow traffic for some interval if possible, since this may indicate that the server cannot accept additional requests at this time.
+
+If the result's [=traffic advice entry/disallowed flag=] is true, then the HTTP server advises that traffic should be disallowed at this time. Establishing new connections and sending requests should be avoided.
+
+Privacy and security considerations {#privacy-and-security}
+===========================================================
+
+<p class="issue">TODO: fill this in</p>
+
+IANA considerations {#iana}
+===========================
+
+Well-known `traffic-advice` URI {#iana-well-known}
+--------------------------------------------------
+
+This document defines well-known URI suffix `traffic-advice` as described by [[WELL-KNOWN]]. It should be submitted for registration as follows:
+
+:  URI suffix
+:: traffic-advice
+:  Change controller
+:: W3C
+:  Specification(s)
+:: This document
+:  Status
+:: provisional
+:  Related information
+:: None
+
+The `application/trafficadvice+json` MIME type {#iana-mime-type}
+----------------------------------------------------------------
+
+This document defines the [=MIME type=] `application/trafficadvice+json` as described by [[RFC6838]]. It should be submitted for registration as follows:
+
+:  Type name
+:: `application`
+:  Subtype name
+:: `trafficadvice+json`
+:  Required parameters
+:: N/A
+:  Optional parameters
+:: N/A
+:  Encoding considerations
+:: Always UTF-8
+:  Security considerations
+:: <p class="issue">TODO: fill this in</p>
+:  Interoperability considerations
+:: <p class="issue">TODO: fill this in</p>
+:  Published specification
+:: This document
+:  Applications that use this media type
+:: HTTP client applications (as noted in the specification, these may not be web browsers)
```

Comment:
Could we just say and cross-link to "traffic advice respecting agents"?

Reply:
Sure, can do. It seemed slightly circular (it more or less becomes "traffic advice is used by programs which use traffic advice") but that's probably okay for now.